lib/src/objects/config_object.cc

```diff
@@ -51,4 +52,42 @@ namespace hocon {
         return config_value_type::OBJECT;
     }
 
+    shared_ptr<const config_mergeable> config_object::with_fallback(shared_ptr<const config_mergeable> mergeable) const {
+        return dynamic_pointer_cast<const config_value>(config_value::with_fallback(mergeable));
```

Comment:
The dynamic_pointer_cast shouldn't be necessary. So we should be able to get rid of the override and fall back to the config_value version.